bentoml/yatai/python_api.py

```diff
@@ -149,3 +162,174 @@ def update_bento_upload_progress(
+                or config().get('aws', 'default_region')
+            }
+            if operator_spec['api_name']:
+                deployment_dict['spec']['aws_lambda_operator_config'][
+                    'api_name'
+                ] = operator_spec['api_name']
+        elif operator_value == DeploymentSpec.GCP_FCUNTION:
+            deployment_dict['spec']['gcp_function_operatorConfig'] = {
+                'region': operator_spec['region']
+                or config().get('google-cloud', 'default_region')
+            }
+            if operator_spec['api_name']:
+                deployment_dict['spec']['gcp_function_operator_config'][
+                    'api_name'
+                ] = operator_spec['api_name']
+        elif operator_value == DeploymentSpec.KUBERNETES:
+            deployment_dict['spec']['kubernetes_operator_config'] = {
+                'kube_namespace': operator_spec['kube_namespace'],
+                'replicas': operator_spec['replicas'],
+                'service_name': operator_spec['service_name'],
+                'service_type': operator_spec['service_type'],
+            }
+        else:
+            raise BentoMLDeploymentException(
+                'Custom deployment is not supported in the current version of BentoML'
+            )
+
+        deployment_yaml = yaml.load(str(deployment_dict))
+
+        yatai_service = get_yatai_service()
+
+        # Make sure there is no active deployment with the same deployment name
+        get_deployment_pb = yatai_service.GetDeployment(
+            GetDeploymentRequest(deployment_name=deployment_name, namespace=namespace)
+        )
+        if get_deployment_pb.status.status_code != status_pb2.Status.NOT_FOUND:
+            raise BentoMLDeploymentException(
+                'Deployment {name} already existed, please use update or apply command'
+                ' instead'.format(name=deployment_name)
+            )
+        return apply_deployment(deployment_yaml)
+    except BentoMLDeploymentException as error:
+        return ApplyDeploymentResponse(status=Status.ABORTED(str(error)))
+    except BentoMLException as error:
+        return ApplyDeploymentResponse(status=Status.INTERNAL(str(error)))
+
+
+# TODO update_deployment is not finished.  It will be working on along with cli command
+def update_deployment(deployment_name, namespace):
+    from bentoml.yatai import get_yatai_service
+
+    yatai_service = get_yatai_service()
+
+    # Make sure there is deployment with the same deployment name
+    get_deployment_pb = yatai_service.GetDeployment(
+        GetDeploymentRequest(deployment_name=deployment_name, namespace=namespace)
+    )
+    if get_deployment_pb.status.status_code == status_pb2.Status.NOT_FOUND:
+        raise BentoMLDeploymentException(
+            'Deployment {name} does not exist, please create deployment first'.format(
+                name=deployment_name
+            )
+        )
+
+
+def apply_deployment(deployment_yaml):
```

Comment:
apply_deployment(deployment_info)

if is yaml

if is dict

if is file_path